src/openrct2/config/Config.cpp

```diff
@@ -837,3 +839,83 @@ bool config_find_or_browse_install_directory()
 
     return true;
 }
+
+std::string FindCsg1datAtLocation(const utf8* path)
+{
+    char buffer[MAX_PATH], checkPath1[MAX_PATH], checkPath2[MAX_PATH];
+    safe_strcpy(buffer, path, MAX_PATH);
+    safe_strcat_path(buffer, "Data", MAX_PATH);
+    safe_strcpy(checkPath1, buffer, MAX_PATH);
+    safe_strcpy(checkPath2, buffer, MAX_PATH);
+    safe_strcat_path(checkPath1, "CSG1.DAT", MAX_PATH);
+    safe_strcat_path(checkPath2, "CSG1.1", MAX_PATH);
+
+    // Since Linux is case sensitive (and macOS sometimes too), make sure we handle case properly.
+    std::string path1result = Path::ResolveCasing(checkPath1);
+    if (!path1result.empty())
+    {
+        return path1result;
+    }
+
+    std::string path2result = Path::ResolveCasing(checkPath2);
+    return path2result;
+}
+
+bool Csg1datPresentAtLocation(const utf8* path)
+{
+    std::string location = FindCsg1datAtLocation(path);
+    return !location.empty();
+}
+
+std::string FindCsg1idatAtLocation(const utf8* path)
+{
+    char checkPath[MAX_PATH];
+    safe_strcpy(checkPath, path, MAX_PATH);
+    safe_strcat_path(checkPath, "Data", MAX_PATH);
+    safe_strcat_path(checkPath, "CSG1I.DAT", MAX_PATH);
+
+    // Since Linux is case sensitive (and macOS sometimes too), make sure we handle case properly.
+    std::string resolvedPath = Path::ResolveCasing(checkPath);
+    return resolvedPath;
+}
+
+bool Csg1idatPresentAtLocation(const utf8* path)
+{
+    std::string location = FindCsg1idatAtLocation(path);
+    return !location.empty();
+}
+
+bool RCT1DataPresentAtLocation(const utf8* path)
+{
+    return Csg1datPresentAtLocation(path) && Csg1idatPresentAtLocation(path) && CsgAtLocationIsUsable(path);
+}
+
+bool CsgIsUsable(rct_gx csg)
+{
+    return csg.header.num_entries >= RCT1_NUM_LL_CSG_ENTRIES;
```

Comment:
```suggestion
    return csg.header.num_entries == RCT1_NUM_LL_CSG_ENTRIES;
```

Exactly 69917 slots are reserved, so this needs to be exact.